src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayload.cs

```diff
@@ -157,6 +191,14 @@ internal enum EventType : int
         Histogram,
         UpDownCounter,
         Error,
+        InstrumentationStarted,
         CounterEnded
     }
+
+    internal enum ErrorType : int
```

Comment:
The latest changes follow the approach you suggested, but I did want to bring one thing up. Some of the error messages (e.g. SharedSessionConfiguredIncorrectly) rely on pieces of the payload that won't be available in `CounterMonitor`, which would make it hard to recreate those messages there without modifying CounterPayload to include the necessary data. Currently, I don't think this is an issue - it appears that only two of the error messages (histograms/timeseries) had dotnet-counters specific messages, so these two are the only ones I have dotnet-counters manually overriding and they don't require any extra payload data. For future errors where this may be an issue, a possible option would be to keep the message generic,  and then dotnet-counters would be able to append specific information onto the end. As an example, instead of this, where the dotnet-counters-specific messaging is in the middle of the message:

`Warning: Time series tracking limit ({_maxTimeSeries}) reached. Not all data is being shown. The limit can be changed with --maxTimeSeries but will use more memory in the target process.`

We could structure it like this:

Generic Version: `Warning: Histogram tracking limit ({configuration.MaxHistograms}) reached. Not all data is being shown. The limit can be changed but will use more memory in the target process.`
DC Version: `Generic Version` + `Use the --maxTimeSeries flag to change the limit.`

Reply:
I think your updated approach looks fine. If, in the future, we need to capture more information to format a custom error in the UI we can always created derived types from ErrorPayload that have more strongly typed data in them. We could also do what you are suggesting of appending some custom text at the end of the generic error message. I expect we'll decide that case-by-case when  new errors come up. For the errors we've got now things look good and our options are open to handle future ones.